src/services/qrService.ts

```diff
@@ -205,19 +198,90 @@ const connectSiopV2 = async (args: IQrDataArgs): Promise<void> => {
       });
       showToast(ToastTypeEnum.TOAST_SUCCESS, {
         title: translate('credentials_share_success_toast_title'),
-        message: format(translate('credentials_share_success_toast_message'), verifier)
+        message: format(translate('credentials_share_success_toast_message'), translateCorrelationIdToName(url.hostname))
       });
     })
     // TODO make human-readable message
     .catch((error: Error) => showToast(ToastTypeEnum.TOAST_ERROR, { message: error.message }));
   };
 
-  RootNavigation.navigate(ScreenRoutesEnum.CREDENTIALS_REQUIRED, {
-    verifier,
-    // TODO currently only supporting 1 presentation definition
-    presentationDefinition: request.presentationDefinitions[0].definition,
-    onSend: async (credentials: Array<VerifiableCredential>) => authenticate(() => onSend(credentials))
-  });
+  const selectRequiredCredentials = async () => {
+    await setTimeout(async () => {
+      const request: VerifiedAuthorizationRequest = await siopGetRequest({...config, id: uuidv4()});
+
+      const contacts = await getContacts({
+        filter: [{
+          identities: {
+            identifier: {
+              correlationId: url.hostname,
+            }
+          }
+        }]
+      })
+      if (contacts.length === 1) {
+        const correlationId = request.payload.client_id
+        const identity: IBasicIdentity = {
+          alias: correlationId,
+          roles: [IdentityRoleEnum.VERIFIER],
+          identifier: {
+            type: CorrelationIdentifierEnum.DID,
+            correlationId
+          }
+        }
+        const hasIdentity = contacts.find((contact: IContact) => contact.identities.some((identity: IIdentity) => identity.identifier.correlationId === correlationId))
+        if (!hasIdentity) {
+          store.dispatch<any>(addIdentity({ contactId: contacts[0].id, identity }))
+        }
+      }
+
+      if (!request.presentationDefinitions || request.presentationDefinitions.length === 0) {
```

Comment:
Not sure what the impact on later code is. But SIOPv2 and OID4VP are seperate. In other words SIOP doesn't require OID4VP. This means that presentation definitions are optional. 

Probably later code assumes we need definitions, but it would be good to at least already provide some TODO's here that this code should be made optional

Reply:
ye i think we need to skip the whole required credentials flow in that case?

added comments